shared/network/README.md

```diff
@@ -0,0 +1,11 @@
+# shared-network
```

Comment:
I think we should take advantage of readmes in shared libraries to explain the purposes and maybe document pieces that otherwise would be lost in code.

Reply:
Yeah, I agree. I'll try to find some time and add some content here.